calibration_assistant.cpp

```diff
@@ -0,0 +1,1250 @@
+            if (fromTimer)
+            {
+                if (fabs(lastDec - dec) > 10)
+                {
+                    if (lastDec != INVALID_DEC)
+                        ShowExplanationMsg(dec);
+                    lastDec = dec;
+                }
+            }
+            else
+                ShowExplanationMsg(dec);
+        }
+    }
+    else
+    {
+        ShowError(_("Mount can't report its pointing position"), true);
+        if (m_pTimer)
+            m_pTimer->Stop();
+    }
+}
+// Return of 'true' means error occurred
+bool CalibrationAssistant::GetCalibPositionRecommendations(int* HA, int* Dec) const
+{
+    double hourAngle;
+    int bestDec;
+    int bestOffset;
+    double lst;
+    double ra;
+    double dec;
+    double eqAltitude;
+    bool pointingEast = false;
+    bool errorSeen = false;
+
+    bestDec = defBestDec;
+    bestOffset = defBestOffset;
+    try
+    {
+        if (pPointingSource && pPointingSource->CanReportPosition())
+        {
+            if (pPointingSource->PreparePositionInteractive())
+                return true;
+            if (pPointingSource->GetCoordinates(&ra, &dec, &lst))
+            {
+                throw ERROR_INFO("CalPositionRecommendations: Mount not reporting pointing position");
+            }
+            hourAngle = norm(lst - ra, -12.0, 12.0);
+            if (hourAngle <= 0)
+                bestOffset = -bestOffset;
+            // Check that we aren't pointing down in the weeds at the default Dec location
+            double lat, lon;
+            if (pPointingSource->GetSiteLatLong(&lat, &lon))
+                bestDec = 0;
+            else if (lat >= 0)
+            {
+                eqAltitude = 90.0 - lat;
+                if (eqAltitude < 30)        // far north location
+                    bestDec += 30 - eqAltitude;
+            }
+            else
+            {
+                eqAltitude = 90 + lat;
+                if (eqAltitude < 30)        // far south location
+                    bestDec -= (30 - eqAltitude);
+            }
+            *HA = bestOffset;
+            *Dec = bestDec;
+            m_pSlewBtn->Enable(pPointingSource->CanSlew());
+        }
+        else
+        {
+            throw ERROR_INFO("CalPositionRecommendations: mount not connected or not reporting pointing info");
+        }
+    }
+    catch (const wxString& msg)
+    {
+        POSSIBLY_UNUSED(msg);
+        *HA = defBestOffset;
+        *Dec = defBestDec;
+        errorSeen = true;
+    }
+    return errorSeen;
+}
+
+void CalibrationAssistant::ShowExplanationMsg(double dec)
+{
+    wxString slewCond;
+    if (pPointingSource->CanSlew())
+        slewCond = _("Use the 'slew' button to move the scope as close as possible to Dec = 0. ");
+    else
+        slewCond = _("Slew the scope as close as possible to Dec = 0. ");
+    if (fabs(dec) > 80)
+    {
+        m_pExplanation->SetLabelText(_("Calibration is likely to fail this close to the pole.\n") + slewCond);

```

Comment:
```suggestion
        m_pExplanation->SetLabelText(_("Calibration is likely to fail this close to the pole.\n") + slewCond);
```
This comment is for all the translatable messages here: let's not include layout formatting ("\n" or trailing " ") in the message that is being translated -- translators may or may not overlook that when they translate and we want the code to be in control of the layout.
Let's ensure that the translators are presented with sentences they can translate without regard to layout formatting. That way they can focus on the translation proper , and the code can do the layout.

Reply:
I've taken a different approach using the 'Wrap' method, much like the profile wizard.  Beyond your points, there's really no way to know how things will appear after translation.  So the long text messages are now wrapped and embedded line feeds have been removed.